keras_cv/models/generative/stable_diffusion/stable_diffusion.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import math
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_cv.models.generative.stable_diffusion.decoder import Decoder
+from keras_cv.models.generative.stable_diffusion.diffusion_model import DiffusionModel
+from keras_cv.models.generative.stable_diffusion.text_encoder import TextEncoder
+from keras_cv.models.generative.stable_diffusion.clip_tokenizer import SimpleTokenizer
+from keras_cv.models.generative.stable_diffusion.constants import (
+    _UNCONDITIONAL_TOKENS,
+    _ALPHAS_CUMPROD,
+)
+
+MAX_PROMPT_LENGTH = 77
+
+
+class StableDiffusion:
```

Comment:
Since we don't have a tutorial yet, maybe a code example in a docstring here would be nice?

Reply:
(especially since this code automagically loads weights, which might be a bit confusing to users)

Reply:
Good call -- I added a full docstring (and an example in `examples/...`)